raiden/network/proxies/token_network.py

```diff
@@ -229,13 +233,7 @@ def new_netting_channel(
                 "the pruning limit. This should not happen."
             )
 
-        channel_exists = self._channel_exists_and_not_settled(
-            participant1=self.node_address,
-            participant2=partner,
-            block_identifier=given_block_identifier,
-        )
-        if channel_exists:
-            raise DuplicatedChannelError("Channel with given partner address already exists")
+        self._raise_if_duplicate_channel(partner=partner, block=given_block_identifier)
```

Comment:
This is missing preconditions checks for:

- New channels cannot be open if [token deposit limit has been reached](https://github.com/raiden-network/raiden-contracts/blob/master/raiden_contracts/data/source/raiden/TokenNetwork.sol#L268)

Reply:
The check above `NoStateForBlockIdentifier` should not exist. It does not prevent the race condition (ValueError has to be handle in `_raise_if_duplicate_channel`), and because at the given block was pruned does not mean the operation will fail (that is why estimate gas is always used).

Reply:
I removed the `raise NoStateForBlockIdentifier` and added a pre/postcondition check for the token network deposit limit.